ide-common/src/main/java/org/digma/intellij/plugin/insights/InsightsProvider.java

```diff
@@ -23,26 +25,32 @@ public class InsightsProvider {
     private static final Logger LOGGER = Logger.getInstance(InsightsProvider.class);
 
     private final AnalyticsService analyticsService;
+    private final DocumentInfoService documentInfoService;
     private final Project project;
 
     private final BuildersHolder buildersHolder = new BuildersHolder();
 
     public InsightsProvider(Project project) {
         analyticsService = project.getService(AnalyticsService.class);
+        documentInfoService = project.getService(DocumentInfoService.class);
         this.project = project;
     }
 
     public InsightsListContainer getInsights(@NotNull MethodInfo methodInfo) {
 
         List<String> objectIds = new ArrayList<>();
         objectIds.add(methodInfo.idWithType());
```

Comment:
I think that you never need id here, only id with type, i'm not sure,ask Shay, but in any case try not so send ids than will never have results, it will probably increase respond time. 
I think you should think in every place when do you need with type and when not and send only the relevant.
I think Shay and Arik can help with that

Reply:
@arik-dig could you please clarify here?
During debugging this part of code I haven't seen any redundant insight ID, maybe I am wrong.

Reply:
I think method `getRelatedCodeObjectIds` was renamed to `getRelatedCodeObjectIdsWithType` so regression works